nvflare/app_opt/lightning/api.py

```diff
@@ -30,6 +31,29 @@
 FL_META_KEY = "__fl_meta__"
 
 
+def load_model_with_te_support(model, state_dict, strict=False):
+    """Load model state dict with support for Transformer Engine's extra state."""
+
+    # Create a lock for thread-safe model loading
+    model_lock = threading.Lock()
+
+    # Filter out _extra_state keys to avoid the UnpicklingError
+    filtered_state_dict = {k: v for k, v in state_dict.items() if "_extra_state" not in k}
+
+    # Use the lock to ensure thread safety during state loading
+    with model_lock:
+        # Initialize TE layers with default extra state if needed
+        for module in model.modules():
+            if hasattr(module, "_extra_state"):
+                if not hasattr(module, "_extra_state") or module._extra_state is None:
+                    module._extra_state = {}
+
+        # Load the state dict with strict=False to ignore missing keys
```

Comment:
this "extra_state" filtering is not general though, other than BioNemo maybe no use case of this, 
we might need to think of a better way to do this filter.

Reply:
It's specific for models trained with [transformer_engine](https://github.com/NVIDIA/TransformerEngine). Do you think it would break other lightning examples?

Reply:
We could make the "_extra_state" variable configurable.

Reply:
Under the new mechanism, could this be written into a NVFlare filter and just register it for the BioNemo examples? since these filtering are not general to other lightning examples

Reply:
You are right. It might be best to remove the extra state as a filter before sending it out to the server. Let me try.

Reply:
I moved this operation to a filter. Works as well.